tests/src/io.etcd/jetcd-core/0.7.5/src/test/java/io/etcd/jetcd/impl/WatchUnitTest.java

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package io.etcd.jetcd.impl;
+
+import io.etcd.jetcd.ByteSequence;
+import io.etcd.jetcd.Client;
+import io.etcd.jetcd.KeyValue;
+import io.etcd.jetcd.Watch;
+import io.etcd.jetcd.api.Event;
+import io.etcd.jetcd.api.Event.EventType;
+import io.etcd.jetcd.api.WatchGrpc.WatchImplBase;
+import io.etcd.jetcd.api.WatchRequest;
+import io.etcd.jetcd.api.WatchResponse;
+import io.etcd.jetcd.common.exception.ClosedClientException;
+import io.etcd.jetcd.common.exception.EtcdException;
+import io.etcd.jetcd.options.WatchOption;
+import io.etcd.jetcd.support.Errors;
+import io.etcd.jetcd.test.GrpcServerExtension;
+import io.etcd.jetcd.watch.WatchEvent;
+import io.grpc.Status;
+import io.grpc.stub.StreamObserver;
+import io_etcd.jetcd_core.impl.TestUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static io_etcd.jetcd_core.impl.TestUtil.bytesOf;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+
+@SuppressWarnings({"SameParameterValue", "ResultOfMethodCallIgnored"})
+// `@org.junit.jupiter.api.Timeout(value = 30)` can't be used in the nativeTest GraalVM CE 22.3
```

Comment:
- I just need to add back the `@Timeout` annotation of Junit at will, and the factory class of Junit will be incorrectly initialized at build time, and this cannot pass the `buildArgs` of `--initialize-at-build-time` to fix. The performance is consistent with the Error Log of the `@EnabledOnOs` annotation mentioned in #24.
- I'm not sure how much help https://github.com/junit-team/junit5/issues/3040 can provide, because that issue seems to have been silent.